uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -3898,6 +3898,147 @@ public function testGetAndGetFullReturnsAllDataWhenLoggedInUserHasEditAllContact
+    );
+
+    $contract2 = HRJobContractFabricator::fabricate(
+      [ 'contact_id' => 2 ],
+      [ 'period_start_date' => '2016-01-01' ]
+    );
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract1['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-02-20'), //Hour will be automatically set to 00:00:00
+      'to_date' =>  CRM_Utils_Date::processDate('2016-02-20'), //Hour will be automatically set to 23:59:59
+    ], true);
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract2['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-02-20 23:59:59'),
+      'to_date' =>  CRM_Utils_Date::processDate('2016-02-23 16:00:00'), //Hour will be automatically set to 23:59:59
+    ], true);
+
+    $result = civicrm_api3('LeaveRequest', 'get', [
+      'from_date' => ['>=' => '2016-02-20'],
+      'sequential' => 1
+    ]);
+    $this->assertEquals(2, $result['count']);
+    $this->assertEquals($contract1['contact_id'], $result['values'][0]['contact_id']);
+    $this->assertEquals($contract2['contact_id'], $result['values'][1]['contact_id']);
+
+    $result = civicrm_api3('LeaveRequest', 'getFull', [
+      'from_date' => ['>=' => '2016-02-20'],
+      'sequential' => 1
+    ]);
+    $this->assertEquals(2, $result['count']);
+    $this->assertEquals($contract1['contact_id'], $result['values'][0]['contact_id']);
+    $this->assertEquals($contract2['contact_id'], $result['values'][1]['contact_id']);
+  }
+
+  public function testGetAndGetFullIncludesAllTheRequestsBetweenTheGivenPeriodAndFromDateAndToDateHaveHours() {
+    $contract = HRJobContractFabricator::fabricate(
+      [ 'contact_id' => 2 ],
+      [ 'period_start_date' => '2016-01-01' ]
+    );
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-03-17 08:00'),
+      'to_date' =>  CRM_Utils_Date::processDate('2016-03-17 09:00:00')
+    ], true);
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-03-17 11:00'),
+      'to_date' =>  CRM_Utils_Date::processDate('2016-03-18 11:00:00')
+    ], true);
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-03-25 17:00:00'),
+      'to_date' =>  CRM_Utils_Date::processDate('2016-03-25 18:00:00'),
+    ], true);
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-03-25 19:00:00'),
+      'to_date' =>  CRM_Utils_Date::processDate('2016-03-25 20:00:00'),
+    ], true);
```

Comment:
`civilint` shows a lot of errors in this class, there are some news ones introduced by this PR such as the lowercase booleans. Do you think it's too much to fix them all in this PR?

Reply:
Yup. I'm sure civilint will complain about pretty much every file in civihr. I would rather fix this only when we finally add style checks to Jenkins (it's in the plans) to avoid having regressions in case people forget to run it manually. When that happens we'll probably have one or a few PRs just for code style fixes.